nncf/common/logging/track_progress.py

```diff
@@ -59,6 +60,65 @@ def render(self, task: "Task") -> Text:
         return Text(text._text[0], style=INTEL_BLUE_COLOR)
 
 
+class WeightedProgress(Progress):
+    """
+    A class to perform a weighted progress tracking.
+    """
+
+    def update(self, task_id: TaskID, **kwargs) -> None:
+        task = self._tasks[task_id]
+
+        advance = kwargs.get("advance", None)
+        if advance is not None:
+            kwargs["advance"] = self.weighted_advance(task, advance)
+
+        completed = kwargs.get("completed", None)
+        if completed is not None:
+            kwargs["completed"] = self.get_weighted_completed(task, completed)
+
+        super().update(task_id, **kwargs)
+
+    def advance(self, task_id: TaskID, advance: float = 1) -> None:
+        if advance is not None:
+            task = self._tasks[task_id]
+            advance = self.weighted_advance(task, advance)
+        super().advance(task_id, advance)
+
+    def reset(self, task_id: TaskID, **kwargs) -> None:
+        task = self._tasks[task_id]
+
+        completed = kwargs.get("completed", None)
+        if completed is not None:
+            kwargs["completed"] = self.get_weighted_completed(task, completed)
+
+        super().reset(task_id, **kwargs)
+
+        if completed == 0:
+            task.fields["completed_steps"] = 0
+
+    @staticmethod
+    def weighted_advance(task: Task, advance: float):
+        """
+        Perform weighted advancement based on an integer step value.
+        """
+        if advance % 1 != 0:
```

Comment:
In which case is it possible?
May be better return 0 in this case, to avoid exception by tracker?

Reply:
> In which case is it possible?

The `update` method is exposed and can be called by a user if tracker is used as a context manager. For example here https://github.com/openvinotoolkit/nncf/blob/develop/tests/post_training/pipelines/image_classification_base.py#L65

> May be better return 0 in this case, to avoid exception by tracker?

I believe it is fine to keep it as exception because the wrong value can only arise in case of wrong usage by a user. Because as I understand, when using it as `for in track(iterable)`, the `advance` argument will always be integer. Also, if something changes inside `rich` and this will no longer be the case, we'll be notified by failed tests.